File: exercise/scala/day07/src/test/scala/WorkshopSpec.scala
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers
import production.*

class WorkshopSpec extends AnyFunSuite with Matchers {
  private val toyName = "1 Super Nintendo"

  test("completing a gift should set its status to produced") {
    val workshop = new Workshop()
    workshop.addGift(Gift(toyName))

    val completedGift = workshop.completeGift(toyName)

    completedGift should not be None
    completedGift.get.status should be(Status.Produced)
  }

  test("completing a non existing gift should return nothing") {
    val workshop = new Workshop()
    val completedGift = workshop.completeGift("UnExisting toy")

    completedGift should be(None)
  }
}

File: exercise/ts/day07/tests/workshop.spec.ts
import {Gift} from "../src/workshop/gift";
import {Workshop} from "../src/workshop/workshop";
import {Status} from "../src/workshop/status";

describe('Workshop', () => {
    const TOY_NAME = '1 Super Nintendo';

    it('completes a gift and sets its status to produced', () => {
        const workshop = new Workshop();
        workshop.addGift(new Gift(TOY_NAME));

        const completedGift = workshop.completeGift(TOY_NAME);

        expect(completedGift).not.toBeNull();
        expect(completedGift!.getStatus()).toBe(Status.Produced);
    });

    it('returns null when trying to complete a non-existing gift', () => {
        const workshop = new Workshop();
        const completedGift = workshop.completeGift('NonExistingToy');

        expect(completedGift).toBeNull();
    });
});
